File: Scala/lambda - closure/1/main.scala
/**
  * Created by fay on 2017/6/1.
  */

object main {
  def main(args: Array[String]): Unit = {
    say(content => {
      println(content)
      "Hello Scala!"
    })
  }

  def say(reply: (String) => String) = {
    println(reply("Hello World!"))
  }
}

File: PHP/lambda - closure/1/main.php
<?php
/**
 * Created by IntelliJ IDEA.
 * User: fay
 * Date: 2017/5/31
 * Time: 下午3:47
 */

function main() {
    say(function ($content) {
        echo $content;
        return "Hello PHP!\n";
    });
}

function say($reply) {
     echo $reply("Hello World!\n");
}

main();
